src/core/unit.c

```diff
@@ -3449,6 +3459,8 @@ int unit_serialize(Unit *u, FILE *f, FDSet *fds, bool serialize_jobs) {
         if (!sd_id128_is_null(u->invocation_id))
                 (void) serialize_item_format(f, "invocation-id", SD_ID128_FORMAT_STR, SD_ID128_FORMAT_VAL(u->invocation_id));
 
+        (void) serialize_item_format(f, "freezer-state", "%d", u->freezer_state);
```

Comment:
This feels like it should be serialised as a string. I mean, if I see "freezer-state: 3", I have no idea what that means.